pkg/utils/cloudconfig/azure/config.go

```diff
@@ -0,0 +1,158 @@
+	ClusterResourceGroup string `json:"clusterResourceGroup,omitempty" mapstructure:"resourceGroup,omitempty"`
+	// name of the virtual network of cluster
+	VnetName string `json:"vnetName,omitempty" mapstructure:"vnetName,omitempty"`
+	// name of the resource group where the virtual network is deployed
+	VnetResourceGroup string `json:"vnetResourceGroup,omitempty" mapstructure:"vnetResourceGroup,omitempty"`
+}
+
+// NewCloudConfigFromFile loads cloud config from a file given the file path.
+func NewCloudConfigFromFile(filePath string) (*CloudConfig, error) {
+	if filePath == "" {
+		return nil, fmt.Errorf("failed to load cloud config: file path is empty")
+	}
+
+	var config CloudConfig
+	configReader, err := os.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open cloud config file: %w, file path: %s", err, filePath)
+	}
+	defer configReader.Close()
+
+	contents, err := io.ReadAll(configReader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read cloud config file: %w, file path: %s", err, filePath)
+	}
+
+	if err := yaml.Unmarshal(contents, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal cloud config: %w, file path: %s", err, filePath)
+	}
+
+	config.trimSpace()
+	if err := config.validate(); err != nil {
+		return nil, fmt.Errorf("failed to validate cloud config: %w, file contents: `%s`", err, string(contents))
+	}
+
+	return &config, nil
+}
+
+// SetUserAgent sets the user agent string to access Azure resources.
+func (cfg *CloudConfig) SetUserAgent(userAgent string) {
+	cfg.UserAgent = userAgent
+}
+
+func (cfg *CloudConfig) validate() error {
+	if cfg.Cloud == "" {
+		return fmt.Errorf("cloud is empty")
+	}
+
+	if cfg.Location == "" {
+		return fmt.Errorf("location is empty")
+	}
+
+	if cfg.SubscriptionID == "" {
+		return fmt.Errorf("subscription ID is empty")
+	}
+
+	if cfg.ResourceGroup == "" {
+		return fmt.Errorf("resource group is empty")
+	}
+
+	if cfg.ClusterResourceGroup == "" {
+		return fmt.Errorf("cluster resource group is empty")
+	}
+
+	if cfg.ClusterName == "" {
+		return fmt.Errorf("cluster name is empty")
+	}
+
+	if cfg.VnetName == "" {
+		return fmt.Errorf("virtual network name is empty")
+	}
+
+	if cfg.VnetResourceGroup == "" {
+		cfg.VnetResourceGroup = cfg.ResourceGroup
+	}
+
+	if !cfg.UseManagedIdentityExtension {
+		if cfg.UserAssignedIdentityID != "" {
+			return fmt.Errorf("useManagedIdentityExtension needs to be true when userAssignedIdentityID is provided")
+		}
+		if cfg.AADClientID == "" || cfg.AADClientSecret == "" {
+			return fmt.Errorf("AAD client ID or AAD client secret is empty")
+		}
+	}
+
+	if cfg.RateLimitConfig == nil {
+		cfg.RateLimitConfig = &RateLimitConfig{CloudProviderRateLimit: true}
+	}
+
+	if cfg.CloudProviderRateLimit {
+		// Assign read rate limit defaults if no configuration was passed in.
+		if cfg.CloudProviderRateLimitQPS == 0 {
+			cfg.CloudProviderRateLimitQPS = consts.RateLimitQPSDefault
```

Comment:
Where do you put these consts?

Reply:
Located here: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/b772f603ea30ea5ebec72e8e0fb4bb3b287111fe/pkg/consts/consts.go#L191